File: client/clojure/src/cdisc_library_api/specs/export_adam_variables_row.clj
(ns cdisc-library-api.specs.export-adam-variables-row
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def export-adam-variables-row-data
  {
   (ds/opt :Version) string?
   (ds/opt :DataStructureName) string?
   (ds/opt :DatasetName) string?
   (ds/opt :VariableGroup) string?
   (ds/opt :VariableName) string?
   (ds/opt :VariableLabel) string?
   (ds/opt :Type) string?
   (ds/opt :CodelistControlledTerms) string?
   (ds/opt :Core) string?
   (ds/opt :CDISCNotes) string?
   })

(def export-adam-variables-row-spec
  (ds/spec
    {:name ::export-adam-variables-row
     :spec export-adam-variables-row-data}))

File: src/main/scala/org/openapitools/client/api/ExportAdamVariablesRow.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ExportAdamVariablesRow._

case class ExportAdamVariablesRow (
  version: Option[String],
dataStructureName: Option[String],
datasetName: Option[String],
variableGroup: Option[String],
variableName: Option[String],
variableLabel: Option[String],
`type`: Option[String],
codelistControlledTerms: Option[String],
core: Option[String],
cDISCNotes: Option[String])

object ExportAdamVariablesRow {
  import DateTimeCodecs._

  implicit val ExportAdamVariablesRowCodecJson: CodecJson[ExportAdamVariablesRow] = CodecJson.derive[ExportAdamVariablesRow]
  implicit val ExportAdamVariablesRowDecoder: EntityDecoder[ExportAdamVariablesRow] = jsonOf[ExportAdamVariablesRow]
  implicit val ExportAdamVariablesRowEncoder: EntityEncoder[ExportAdamVariablesRow] = jsonEncoderOf[ExportAdamVariablesRow]
}
